seed/data_importer/meters_parser.py

```diff
@@ -217,19 +209,27 @@ def _get_property_id_from_source(self, source_id, shared_details):
         """
         property_id = self._source_to_property_ids.get(source_id, None)
 
+        # If the PM ID has been previously found to be unlinkable, return False
+        if source_id in self._unlinkable_pm_ids:
+            return False
+
         if property_id is not None:
             shared_details['property_id'] = property_id
         else:
             try:
                 """
-                Filter used because multiple PropertyStates will be found, but the
-                underlying Property should be the same, so take the first.
+                QuerySet filters() used to find resulting property_id because
+                multiple PropertyStates can be found, but the underlying
+                Property amongst them should be the same. So take the first.
                 """
-                property_id = PropertyState.objects \
-                    .filter(pm_property_id__exact=source_id, organization_id__exact=self._org_id)[0] \
-                    .propertyview_set \
-                    .first() \
-                    .property_id
+                possible_matches = PropertyState.objects.filter(
+                    pm_property_id__exact=source_id,
+                    organization_id__exact=self._org_id
+                )
+
+                property_id = PropertyView.objects.filter(
```

Comment:
@axelstudios This is the fix for the improper filtering done previously (from Slack): 
> [The old logic] doesn’t filter out PropertyStates with data_state=DATA_STATE_DELETE (which are never associated to PropertyViews). 

The corresponding test for this can be found here: https://github.com/SEED-platform/seed/pull/1907/files#diff-b35140d6e17c6cc4ee66b2f453a361f3R549